### PR TITLE
chore(samples): Add samples for Cloud Spanner Scheduled Backups

### PR DESCRIPTION
**Pull request by ketanv3**
## Description

This PR adds code samples for the following APIs:

1. CreateBackupSchedule (both full and incremental backup schedules)
2. ListBackupSchedules
3. GetBackupSchedule
4. UpdateBackupSchedule
5. DeleteBackupSchedule

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	"fmt"
+	"io"
+	"time"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"google.golang.org/protobuf/types/known/durationpb"
+)
+
+func createFullBackupSchedule(ctx context.Context, w io.Writer, dbName string, scheduleId string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go:30]: issue: don't pass context objects around, including code samples.

File: spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	res, err := client.CreateBackupSchedule(ctx, &req)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Created full backup schedule: %s", res)
+	return nil
+}
+
+func createIncrementalBackupSchedule(ctx context.Context, w io.Writer, dbName string, scheduleId string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go:65]: issue: move this sample to a separate file.

See: https://googlecloudplatform.github.io/samples-style-guide/#one-per-file

Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go:65]: Same as above: don't pass in a context object.

File: spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	client, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	req := databasepb.CreateBackupScheduleRequest{
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go:72]: Same as above: provide a comment explaining what this code actually does.

File: spanner/spanner_snippets/spanner/spanner_delete_backup_schedule.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+)
+
+func deleteBackupSchedule(ctx context.Context, w io.Writer, dbName string, scheduleId string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_delete_backup_schedule.go:28]: Same as above: don't pass in context object.

File: spanner/spanner_snippets/spanner/spanner_get_backup_schedule.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"context"
+	"fmt"
+	"io"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+)
+
+func getBackupSchedule(ctx context.Context, w io.Writer, dbName string, scheduleId string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_get_backup_schedule.go:28]: Same as previous: don't pass in context object.

File: spanner/spanner_snippets/spanner/spanner_list_backup_schedules.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"google.golang.org/api/iterator"
+)
+
+func listBackupSchedules(ctx context.Context, w io.Writer, dbName string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_list_backup_schedules.go:29]: Same as previous: don't pass in context object.

File: spanner/spanner_snippets/spanner/spanner_update_backup_schedule.go [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+	"io"
+	"time"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+func updateBackupSchedule(ctx context.Context, w io.Writer, dbName string, scheduleId string) error {
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_update_backup_schedule.go:31]: Same as previous: don't pass in context object.

File: spanner/spanner_snippets/spanner/spanner_update_backup_schedule.go [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+	client, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	req := databasepb.UpdateBackupScheduleRequest{
```
Comment (telpirion) [inline spanner/spanner_snippets/spanner/spanner_update_backup_schedule.go:38]: Same as above: Add a comment explaining what this request does.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go (hunk 1 above)
Comment (ketanv3) [reply spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go]: Fixed!

File: spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go (hunk 2 above)
Comment (ketanv3) [reply spanner/spanner_snippets/spanner/spanner_create_backup_schedule.go]: Fixed!

**Review by telpirion — APPROVED** [review]
Looks great! Thank you.

**Comment by ketanv3** [comment]
Thanks for the quick review, @telpirion!

You may want to retry the Go/Build GitHub action as it failed due to a transient network error.

**Pull request merged**